crates/stackable-operator/src/crd/git_sync/v1alpha1_impl.rs

```diff
@@ -0,0 +1,798 @@
+        resolved_product_image: &ResolvedProductImage,
+        git_sync: &GitSync,
+        one_time: bool,
+        env_vars: &[EnvVar],
+        volume_mounts: &[VolumeMount],
+    ) -> Result<k8s_openapi::api::core::v1::Container, Error> {
+        let container = ContainerBuilder::new(container_name)
+            .context(InvalidContainerNameSnafu)?
+            .image_from_product_image(resolved_product_image)
+            .command(vec![
+                "/bin/bash".to_string(),
+                "-x".to_string(),
+                "-euo".to_string(),
+                "pipefail".to_string(),
+                "-c".to_string(),
+            ])
+            .args(vec![Self::create_git_sync_command(git_sync, one_time)])
+            .add_env_vars(env_vars.into())
+            .add_volume_mounts(volume_mounts.to_vec())
+            .context(AddVolumeMountSnafu)?
+            .resources(
+                ResourceRequirementsBuilder::new()
+                    .with_cpu_request("100m")
+                    .with_cpu_limit("200m")
+                    .with_memory_request("64Mi")
+                    .with_memory_limit("64Mi")
+                    .build(),
+            )
+            .build();
+        Ok(container)
+    }
+
+    fn create_git_sync_command(git_sync: &GitSync, one_time: bool) -> String {
+        let internal_args = [
+            Some(("--repo".to_string(), git_sync.repo.to_owned())),
+            Some(("--ref".to_string(), git_sync.branch.to_owned())),
+            Some(("--depth".to_string(), git_sync.depth.to_string())),
+            Some((
+                "--period".to_string(),
+                format!("{}s", git_sync.wait.as_secs()),
+            )),
+            Some(("--link".to_string(), GIT_SYNC_LINK.to_string())),
+            Some(("--root".to_string(), GIT_SYNC_ROOT_DIR.to_string())),
+            one_time.then_some(("--one-time".to_string(), "true".to_string())),
+        ]
+        .into_iter()
+        .flatten()
+        .collect::<BTreeMap<_, _>>();
+
+        let internal_git_config = [(
+            GIT_SYNC_SAFE_DIR_OPTION.to_string(),
+            GIT_SYNC_ROOT_DIR.to_string(),
+        )]
+        .into_iter()
+        .collect::<BTreeMap<_, _>>();
+
+        let mut user_defined_args = BTreeMap::new();
+        // The key and value in Git configs are separated by a colon, but both
+        // can contain either escaped colons or unescaped colons if enclosed in
+        // quotes. To avoid parsing, just a vector is used instead of a map.
+        let mut user_defined_git_configs = Vec::new();
+
+        for (key, value) in &git_sync.git_sync_conf {
+            // The initial git-sync implementation in the airflow-operator
+            // (https://github.com/stackabletech/airflow-operator/pull/381)
+            // used this condition to find Git configs. It is also used here
+            // for backwards-compatibility:
+            if key.to_lowercase().ends_with("-git-config") {
```

Comment:
note: This condition seems a little weird. As far as I can see, the only way to provide custom git configs in git-sync is to use `--git-config`. Why don't we instead use the following expression:

```rust
if key.to_lowercase() == "--git-config" {}
```

Independent of which approach we finally go with, the value we compare against should live in a constant.

Reply:
I'm pretty sure that in an earlier version of gitsync it was possible to pass `-git-config` or `--git-config` (at least, I made a comment to that effect). However, I can't find any place in the code to confirm that, and it is certainly not the case in the 4.x version, so we can tighten that up as per this suggestion.

Reply:
80d7b3e5a421c2cd171abce68aa41bab1b6143c7